pandas/core/dtypes/common.py

```diff
@@ -1397,7 +1397,7 @@ def is_bool_dtype(arr_or_dtype) -> bool:
         # guess this
         return arr_or_dtype.is_object and arr_or_dtype.inferred_type == "boolean"
     elif is_extension_array_dtype(arr_or_dtype):
-        return getattr(arr_or_dtype, "dtype", arr_or_dtype)._is_boolean
+        return getattr(dtype, "_is_boolean", False)
```

Comment:
comment that this is de facto an `isinstance(dtype, BooleanDtype)`?

Reply:
I think this might also return True for sparse and categorical.